surface/include/pcl/surface/impl/marching_cubes.hpp

```diff
@@ -59,26 +52,17 @@ pcl::MarchingCubes<PointNT>::~MarchingCubes ()
 template <typename PointNT> void
 pcl::MarchingCubes<PointNT>::getBoundingBox ()
 {
-  pcl::getMinMax3D (*input_, min_p_, max_p_);
-
-  min_p_ -= (max_p_ - min_p_) * percentage_extend_grid_/2;
-  max_p_ += (max_p_ - min_p_) * percentage_extend_grid_/2;
-
-  Eigen::Vector4f bounding_box_size = max_p_ - min_p_;
-
-  bounding_box_size = max_p_ - min_p_;
-  PCL_DEBUG ("[pcl::MarchingCubesHoppe::getBoundingBox] Size of Bounding Box is [%f, %f, %f]\n",
-             bounding_box_size.x (), bounding_box_size.y (), bounding_box_size.z ());
-  double max_size =
-      (std::max) ((std::max)(bounding_box_size.x (), bounding_box_size.y ()),
-          bounding_box_size.z ());
-  (void)max_size;
-  // ????
-  //  data_size_ = static_cast<uint64_t> (max_size / leaf_size_);
-  PCL_DEBUG ("[pcl::MarchingCubesHoppe::getBoundingBox] Lower left point is [%f, %f, %f]\n",
-             min_p_.x (), min_p_.y (), min_p_.z ());
-  PCL_DEBUG ("[pcl::MarchingCubesHoppe::getBoundingBox] Upper left point is [%f, %f, %f]\n",
-             max_p_.x (), max_p_.y (), max_p_.z ());
+  PointNT max_pt, min_pt;
+  pcl::getMinMax3D (*input_, min_pt, max_pt);
+
+  lower_boundary_ = min_pt.getVector3fMap ().array ();
+  upper_boundary_ = max_pt.getVector3fMap ().array ();
```

Comment:
You should use [`getArray3fMap()`](http://docs.pointclouds.org/trunk/point__types_8hpp_source.html#l00157) instead.